File: baekjoon/4000/4673.java
public class Main {
    public static int SumOfDigits(int n) {
        int sum = 0;
        while(n > 0) {
            sum += (n%10);
            n /= 10;
        }
        return sum;
    }
    
    public static void main(String[] args) {
        int[] array = new int[10001];
        for(int i=1; i<array.length; i++) {
            if(array[i] == 0)
                System.out.println(i);
            int index = i + Main.SumOfDigits(i);
            if(index <= 10000)
                array[index]++;
        }
    }
}

File: baekjoon/4000/4673.rs
fn main() {
    use std::io::Write;
    let out = std::io::stdout();
    let mut out = std::io::BufWriter::new(out.lock());
    let mut has_parent = [false; 10001];
    for mut x in 1..=10000 {
        let mut sum = x;
        while x != 0 {
            sum += x % 10;
            x /= 10;
        }
        if sum <= 10000 {
            has_parent[sum] = true;
        }
    }
    has_parent[0] = true;
    for (i, &x) in has_parent.iter().enumerate() {
        if !x {
            writeln!(out, "{}", i).ok();
        }
    }
}
